frontend/src/app/components/Chrome/Sidebar/NavLinks.tsx

```diff
@@ -0,0 +1,39 @@
+import React, { Component } from 'react';
+import { Link } from "react-router-dom";
+
+export default class NavLinks extends Component {
+    public state = {
+        id: 'userID',
+        isLoggedIn: false,
+        moduleText: '',
+        searchText: 'Search'
+    };
+
+    public render() {
+        if (!this.state.isLoggedIn) {
+            fetch("/system/sling/info.sessionInfo.json")
+                .then(response => response.json())
+                .then(responseJSON => {
+                    if (responseJSON[this.state.id] !== 'anonymous') {
+                        this.setState({ moduleText: 'New Module' })
+                        this.setState({ isLoggedIn: true })
+                    }
+                })
+        }
+
+        return (
+            <React.Fragment>
+                <li>
+                    <Link to='/search'>
+                        {this.state.searchText}
+                    </Link>
+                </li>
+                <li>
```

Comment:
Would it make sense to change this to `{this.state.moduleText.length > 0 && <li>...`?

Reply:
maybe I am not sure if it makes a difference performance wise, but I don't think it would hurt us either way. So I'll add it.